### ISSUE TITLE
Doubt in score function

### ISSUE THREAD

**Issue by vardaan123** [awslabs/dgl-ke#108]
Thanks for this awesome library! 
I have a confusion in the score function for the TransE model which the library uses. The edge score is given in https://github.com/awslabs/dgl-ke/blob/master/python/dglke/models/pytorch/score_fun.py#L59 as (gamma - norm(h+r-t)). In the forward() method of the KE model https://github.com/awslabs/dgl-ke/blob/master/python/dglke/models/general_models.py#L346, the final score is the sum of logsigmoid of positive score and mean of logsigmoid of negative score. Could you please point me to a paper that uses this particular loss function. It seems to be a combination of margin based loss and logistic loss.

**Comment by classicsong** [comment]
For https://github.com/awslabs/dgl-ke/blob/master/python/dglke/models/general_models.py#L346, The predict_neg_score is used by training (we need to record the grad) and inference (no need for grad)

**Comment by vardaan123** [comment]
Thanks for your response. It seems that the line numbers referred by me have changed since. My doubt is about the score function being used in simple TransE model. In section 2 (Background) of your paper, you have mentioned the logistic loss and pairwise ranking loss but in code, your score function seems to be a combination of both. I am interested in the mathematical expression of your score function for the TransE model.

**Comment by classicsong** [comment]
> Thanks for your response. It seems that the line numbers referred by me have changed since. My doubt is about the score function being used in simple TransE model. In section 2 (Background) of your paper, you have mentioned the logistic loss and pairwise ranking loss but in code, your score function seems to be a combination of both. I am interested in the mathematical expression of your score function for the TransE model.

The pairwise ranking loss is in the plan #81
